pkg/sdk/external_tables_impl.go

```diff
@@ -0,0 +1,99 @@
+package sdk
+
+import "context"
+
+var (
+	_ ExternalTables                = (*externalTables)(nil)
+	_ convertibleRow[ExternalTable] = (*externalTableRow)(nil)
+)
+
+type externalTables struct {
+	client *Client
+}
+
+func (v *externalTables) Create(ctx context.Context, req *CreateExternalTableRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) CreateWithManualPartitioning(ctx context.Context, req *CreateWithManualPartitioningExternalTableRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) CreateDeltaLake(ctx context.Context, req *CreateDeltaLakeExternalTableRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) CreateUsingTemplate(ctx context.Context, req *CreateExternalTableUsingTemplateRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) Alter(ctx context.Context, req *AlterExternalTableRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) AlterPartitions(ctx context.Context, req *AlterExternalTablePartitionRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) Drop(ctx context.Context, req *DropExternalTableRequest) error {
+	return validateAndExec(v.client, ctx, req.toOpts())
+}
+
+func (v *externalTables) Show(ctx context.Context, req *ShowExternalTableRequest) ([]ExternalTable, error) {
+	dbRows, err := validateAndQuery[externalTableRow](v.client, ctx, req.toOpts())
+	if err != nil {
+		return nil, err
+	}
+	resultList := convertRows[externalTableRow, ExternalTable](dbRows)
+	return resultList, nil
+}
+
+func (v *externalTables) ShowByID(ctx context.Context, req *ShowExternalTableByIDRequest) (*ExternalTable, error) {
+	if !validObjectidentifier(req.id) {
+		return nil, ErrInvalidObjectIdentifier
+	}
+
+	externalTables, err := v.client.ExternalTables.Show(ctx, NewShowExternalTableRequest().WithLike(String(req.id.Name())))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, t := range externalTables {
+		if t.ID() == req.id {
+			return &t, nil
+		}
+	}
+
+	return nil, ErrObjectNotExistOrAuthorized
+}
+
+func (v *externalTables) DescribeColumns(ctx context.Context, req *DescribeExternalTableColumnsRequest) ([]ExternalTableColumnDetails, error) {
+	rows, err := validateAndQuery[externalTableColumnDetailsRow](v.client, ctx, &describeExternalTableColumns{
+		name: req.id,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]ExternalTableColumnDetails, len(rows))
```

Comment:
why not `convert` method also here and in the next method? You have different rows, so you can support entirely different convert methods and reuse the function instead of copy-pasting.

Reply:
Used convert with convertRows